hail/src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -787,6 +787,12 @@ object Interpret {
         function.execute(ctx, child.execute(ctx))
       case BlockMatrixToValueApply(child, function) =>
         function.execute(ctx, child.execute(ctx))
+      case BlockMatrixCollect(child) =>
+        val bm = child.execute(ctx)
+        val breezeMat = bm.toBreezeMatrix()
+        val shape = IndexedSeq(bm.nRows, bm.nCols)
+        // transpose because breeze toArray is column major
+        SafeNDArray(shape, breezeMat.t.toArray)
```

Comment:
If there's no way to avoid an actual transpose, it would be better to do it before collecting to breeze, so you're only transposing the blocks.

Reply:
In breeze transpose is just a `isTransposed` flag, so this should just be telling it which way to collect.

Reply:
Right, but then the collect is the transpose. You're collecting a column major matrix in row major order, which for larger matrices will have terrible cache behavior.

Reply:
Made the change anyway.